fenix/app/src/test/java/org/mozilla/fenix/components/metrics/DefaultMetricsStorageTest.kt

```diff
@@ -382,6 +384,120 @@ class DefaultMetricsStorageTest {
+        val currentTime = System.currentTimeMillis()
+        installTime = currentTime - (dayMillis * 5)
+        every { settings.adjustUseCount.value } returns 3
+        every { settings.adjustEarlySearchCount.value } returns 1
+        every { settings.adjustUserActivatedSent } returns true
+
+        val result = storage.shouldTrack(Event.GrowthData.UserActivated(fromSearch = false))
+
+        assertFalse(result)
+    }
+
+    @Test
+    fun `WHEN first week usage signal is sent a full day after last sent THEN settings will be updated accordingly`() = runTest(dispatcher) {
+        val captureSent = slot<Long>()
+        val currentTime = System.currentTimeMillis()
+        installTime = currentTime - (dayMillis * 3)
+        every { settings.adjustUseCount.value } returns 1
+        every { settings.adjustUseCount.increment() } just Runs
+        every { settings.adjustUseCountLastIncrement } returns 0L
+        every { settings.adjustUseCountLastIncrement = capture(captureSent) } returns Unit
+
+        storage.updatePersistentState(Event.GrowthData.UserActivated(fromSearch = false))
+
+        assertTrue(captureSent.captured > 0L)
+    }
+
+    @Test
+    fun `WHEN first week usage signal is sent less than a full day after last sent THEN settings will not be updated`() = runTest(dispatcher) {
+        val captureSent = slot<Long>()
+        val currentTime = System.currentTimeMillis()
+        installTime = currentTime - (dayMillis * 3)
+        val lastUsageIncrementTime = currentTime - (dayMillis / 2)
+        every { settings.adjustUseCount.value } returns 1
+        every { settings.adjustUseCountLastIncrement } returns lastUsageIncrementTime
+        every { settings.adjustUseCountLastIncrement = capture(captureSent) } returns Unit
+
+        storage.updatePersistentState(Event.GrowthData.UserActivated(fromSearch = false))
+
+        assertFalse(captureSent.isCaptured)
+    }
+
+    @Test
+    fun `WHEN first week search activity is sent in second half of first week THEN settings will be updated`() = runTest(dispatcher) {
+        val captureSent = slot<Long>()
+        val currentTime = System.currentTimeMillis()
+        installTime = currentTime - (dayMillis * 3) + 1
+        every { settings.adjustEarlySearchCount.value } returns 0
+        every { settings.adjustEarlySearchCount.increment() } just Runs
+        every { settings.adjustEarlySearchCountLastIncrement } returns 0L
+        every { settings.adjustEarlySearchCountLastIncrement = capture(captureSent) } returns Unit
+
+        storage.updatePersistentState(Event.GrowthData.UserActivated(fromSearch = true))
+
+        assertTrue(captureSent.captured > 0L)
+    }
+
+    @Test
+    fun `WHEN first week search activity is sent in first half of first week THEN settings will not be updated`() = runTest(dispatcher) {
+        val captureSent = slot<Long>()
+        val currentTime = System.currentTimeMillis()
+        installTime = currentTime - (dayMillis * 2) + 1
```

Comment:
Clock consistency might make this impossible, but I'm wondering if this could be (dayMillis * 3) - 1 to align with the test above it.

Reply:
I had actually constructed these tests a little wrong and aligning the `installTime` to a similar length revealed my error.
Fixed the tests to be more stable.